core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -183,6 +185,11 @@ class KafkaApis(val requestChannel: RequestChannel,
     val updateMetadataResponse =
       if (authorize(request.session, ClusterAction, Resource.ClusterResource)) {
         replicaManager.maybeUpdateMetadataCache(correlationId, updateMetadataRequest, metadataCache)
+        if (controller.isActive && !updateMetadataRequest.partitionStates.isEmpty) {
```

Comment:
A benefit to having the check is preventing unnecessary iterations over the updateMetadataRequest topics and calls to the topic purgatory in brokers that aren't the controller on each updateMetadataRequest.